### PR TITLE
TE-2.1: gRIBI IPv4 Entry - Added interface type ethernetCsmacd 

### PR DESCRIPTION
**Pull request by stthummuri**
1. Added interface type ethernetCsmacd to setDutInterfaceWithState

2. Ported changes from https://github.com/openconfig/featureprofiles/pull/912/ to OTG location

### CODE REVIEW

**Review by octpetre — APPROVED** [review]

File: feature/gribi/otg_tests/ipv4_entry_test/ipv4_entry_test.go [hunk 1]
```diff
@@ -503,6 +509,8 @@ func setDUTInterfaceWithState(t testing.TB, dut *ondatra.DUTDevice, dutPort *att
 	dc := gnmi.OC()
 	i := &oc.Interface{}
 	i.Enabled = ygot.Bool(state)
+	i.Type = ethernetCsmacd
+	i.Name = ygot.String(p.Name())
 	gnmi.Update(t, dut, dc.Interface(p.Name()).Config(), i)
```
Comment (sachendras) [inline feature/gribi/otg_tests/ipv4_entry_test/ipv4_entry_test.go:514]: This function does not exist in the corresponding ATE test, please verify

Comment (sachendras) [reply feature/gribi/otg_tests/ipv4_entry_test/ipv4_entry_test.go:514]: The function seems necessary for OTG

**Review by sachendras — APPROVED** [review]

**Pull request merged**